Objects/dictobject.c

```diff
@@ -4285,3 +4479,970 @@ _PyDictKeys_DecRef(PyDictKeysObject *keys)
 {
     DK_DECREF(keys);
 }
+
```

Comment:
The implementation of dict already is complex. I think it would be better to keep the implementation of OrderedDict in separate file. In additional to long-term effects this makes making review much harder.

Reply:
I created split version of this pullrequest: methane/cpython#9

As you can see, many private functions has "extern" linkage to be used from `odictobject.c`.
https://github.com/methane/cpython/pull/9/files#diff-b06f0b32d4365b3d104057b8d40c90f8R96

@serhiy-storchaka @rhettinger Which do you prefer?